modules/diskq/logqueue-disk-non-reliable.c

```diff
@@ -312,14 +319,18 @@ _pop_head(LogQueue *s, LogPathOptions *path_options)
     }
 
 success:
-  _maybe_move_messages_among_queue_segments(self);
+  if (!_maybe_move_messages_among_queue_segments(self))
+    {
+      stats_update = FALSE;
```

Comment:
I don't understand this commit. Could you please explain it?

My understanding is that `_pop_head()` should decrement the queued messages counter whenever a message is returned from this function.

AFAIK `_maybe_move_messages_among_queue_segments()` is a transparent side effect, and it should not change the way we calculate queued messages.

Reply:
I get it now: the move operation can really fail, hence the conditional stats update.